salt/modules/yumpkg.py

```diff
@@ -658,13 +637,29 @@ def list_upgrades(refresh=True, **kwargs):
     exclude_arg = _get_excludes_option(**kwargs)
 
     if salt.utils.is_true(refresh):
-        refresh_db(**kwargs)
+        refresh_db(check_update=False, **kwargs)
+
+    cmd = [_yum(), '--quiet']
+    cmd.extend(repo_arg)
+    cmd.extend(exclude_arg)
+    cmd.extend(['list', 'updates'])
+    out = __salt__['cmd.run_all'](cmd,
+                                  output_loglevel='trace',
+                                  ignore_retcode=True,
+                                  python_shell=False)
+    if out['retcode'] != 0 and 'Error:' in out:
+        return {}
+    # Find end of first line so we can skip it
+    header_end = out['stdout'].find('\n')
+    if header_end == -1:
+        return {}
 
-    repoquery_args = repo_arg + exclude_arg
-    repoquery_args.extend(['--all', '--pkgnarrow=updates'])
-    updates = _repoquery_pkginfo(repoquery_args)
+    updates = _yum_pkginfo(out['stdout'][header_end + 1:])
     return dict([(x.name, x.version) for x in updates])
 
+# Preserve expected CLI usage (yum list updates)
+list_updates = list_upgrades
```

Comment:
Interesting. I suppose we've never really strictly defined the purpose of `__func_alias__`. To the best of my recollection, I've seen it used for both. Happy to leave this in of course. It's not a big deal at all. I'll go back through the history of that implementation and see where my possible misunderstanding came from. Thanks, @terminalmage